crates/exex/src/model.rs

```diff
@@ -0,0 +1,192 @@
+use cairo_vm::{types::relocatable::MaybeRelocatable, Felt252};
+use serde::{Deserialize, Serialize};
+
+/// A custom wrapper around [`MaybeRelocatable`] for the Keth execution environment.
+///
+/// This struct serves as a utility for wrapping [`MaybeRelocatable`] values in Keth, which uses
+/// [`Felt252`] as the core type to represent numerical values within its system.
+///
+/// Additionally, this wrapper facilitates operations such as conversions between different Keth
+/// types, making it easier to work with the underlying data structures.
+///
+/// # Usage
+/// - This type is primarily used for wrapping values in Keth, enabling smooth interoperation
+///   between [`Felt252`], [`MaybeRelocatable`] and reth primitive data.
+#[derive(Debug, Eq, Ord, Hash, PartialEq, PartialOrd, Clone, Serialize, Deserialize)]
+pub struct KethMaybeRelocatable(MaybeRelocatable);
+
+impl Default for KethMaybeRelocatable {
+    fn default() -> Self {
+        Self::zero()
+    }
+}
+
+impl KethMaybeRelocatable {
+    /// Creates a [`KethMaybeRelocatable`] instance representing the value `0`.
+    ///
+    /// This method wraps [`Felt252::ZERO`] in the [`MaybeRelocatable`] type and
+    /// provides a convenient way to represent zero in computations or storage.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use kakarot_exex::model::KethMaybeRelocatable;
+    ///
+    /// let zero_value = KethMaybeRelocatable::zero();
+    /// ```
+    pub fn zero() -> Self {
+        Self(Felt252::ZERO.into())
+    }
+
+    /// Creates a [`KethMaybeRelocatable`] instance representing the value `1`.
+    ///
+    /// This method wraps [`Felt252::ONE`] in the [`MaybeRelocatable`] type and
+    /// provides a convenient way to represent one in computations or storage.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use kakarot_exex::model::KethMaybeRelocatable;
+    ///
+    /// let one_value = KethMaybeRelocatable::one();
+    /// ```
+    pub fn one() -> Self {
+        Self(Felt252::ONE.into())
+    }
+}
+
+impl From<Felt252> for KethMaybeRelocatable {
+    fn from(value: Felt252) -> Self {
+        Self(value.into())
+    }
+}
+
+impl From<u64> for KethMaybeRelocatable {
+    fn from(value: u64) -> Self {
+        Self(Felt252::from(value).into())
+    }
+}
+
+/// [`KethOption`] is a custom representation of a Rust [`Option<u64>`] type, specifically tailored
```

Comment:
can't we make this `Option<T: Into<KethMaybeRelocatable>>` ?

Reply:
Yes good idea, just fixed